python/lsst/daf/butler/core/formatter.py

```diff
@@ -86,24 +82,18 @@ def getFormatter(self, storageClass, datasetType=None):
         ----------
         storageClass : `StorageClass`
             Get `Formatter` associated with this `StorageClass`, unless.
-        datasetType : `DatasetType` or `str, optional
+        datasetType : `DatasetType` or `str`, optional
             If given, look if an override has been specified for this `DatasetType` and,
             if so return that instead.
         """
-        if datasetType:
-            try:
-                typeName = self._registry[self._getName(datasetType)]()
-            except KeyError:
-                pass
-        typeName = self._registry[self._getName(storageClass)]
-        return self._getInstanceOf(typeName)
+        return self.getFromRegistry(storageClass, override=datasetType)
```

Comment:
I can switch this to using composition if you prefer since the get and put methods do not have the same names as the base class implementation

Reply:
Yes, that way we can also get rid of my other gripe (having `refType` specified) and move that to an argument (e.g. `self.mappingFactory = MappingFactory(Formatter)`).